lib/cli.js

```diff
@@ -98,7 +99,15 @@ if (options.debug) {
     debug('cli.input', cli.input);
 }
 
-npmCheck(options)
+(
+    options.installer === 'auto' ?
+        detectPreferredInstaller(options.cwd) :
+        Promise.resolve(process.env.NPM_CHECK_INSTALLER)
```

Comment:
`process.env.NPM_CHECK_INSTALLER` should already have been read into `options.installer`, this will override any specific installer given by a CLI flag.